lib/purescript-cst/src/Language/PureScript/Names.hs

```diff
@@ -73,6 +79,10 @@ data Ident
   -- A generated name used only for type-checking
   --
   | UnusedIdent
+  -- |
+  -- A generated name used only for internal transformations
+  --
+  | InternalIdent InternalIdentData
```

Comment:
> I should add a comment on InternalIdentData to this effect, but my intent was that if we wanted to do another transformation we would just add constructors to InternalIdentData. InternalIdent is a way of encapsulating all the weird local pseudo-names away from unrelated functions that match on Ident; those functions can ignore the InternalIdentData argument and so adding new InternalIdentData constructors will be a smaller code change than adding new Ident constructors.

Gotcha. Let's leave it as is.

> Constructing InternalIdent RuntimeLazyFactory certainly shouldn't, and in Ident txt -> InternalIdent $ Lazy txt, I don't think that txt could possibly carry a thunk associated with our transformation since we never create new Ident _ values? But if you think it's best I have no issue with banging these constructors up anyway.

I think we should add a bang here. Adding them won't do any harm, right? And if it is needed, then it solves this, right?